nmostesting/ControllerTest.py

```diff
@@ -0,0 +1,560 @@
+
+        del_url = self.mock_registry_base_url + 'x-nmos/registration/v1.3/resource/receivers/' + receiver['id']
+
+        valid, r = self.do_request("DELETE", del_url)
+        if not valid:
+            # Hmm - do we need these exceptions as the registry is our own mock registry?
+            raise NMOSTestException(test.FAIL(test, "Registration API returned an unexpected response: {}".format(r)))
+
+        del_url = self.mock_registry_base_url + 'x-nmos/registration/v1.3/resource/devices/' + receiver['device_id']
+
+        valid, r = self.do_request("DELETE", del_url)
+        if not valid:
+            # Hmm - do we need these exceptions as the registry is our own mock registry?
+            raise NMOSTestException(test.FAIL(test, "Registration API returned an unexpected response: {}".format(r)))
+
+    def pre_tests_message(self):
+        """
+        Introduction to Controller Test Suite
+        """
+        # In order to give the tests some context, a pre tests message is displayed
+        # on the Testing Façade prior to the tests starting. This communicates any
+        # pre-requisites or setup required by the Test User. The pre tests message
+        # will vary depending on whether or not unicast DNS-SD is being used to discover the mock Registry.
+
+        dns_sd_enabled = CONFIG.ENABLE_DNS_SD and CONFIG.DNS_SD_MODE == "unicast"
+
+        paragraphs = []
+        paragraphs.append('These tests validate an NMOS Controller under Test (NCuT).\n\n')
+
+        paragraphs.append('A Test AMWA IS-04 reference Registry is available on the network')
+
+        paragraphs.append(' and is being advertised via unicast DNS-SD.\n\n' if dns_sd_enabled else '.\n\n')
+
+        paragraphs.append(
+            'Please ensure that the following configuration has been set on the NCuT machine.\n\n '
+            '* Ensure that the primary DNS of the NCuT machine has been set to \"' + get_default_ip() + '\". \n'
+            '* Ensure that the NCuT unicast search domain is set to \"' + CONFIG.DNS_DOMAIN + '\". \n\n'
+            'Alternatively it ' if dns_sd_enabled else 'It ')
+
+        paragraphs.append(
+            'is possible to reach the Registry via the following URL:\n\n' + self.mock_registry_base_url +
+            'x-nmos/query/v1.3\n\n'
+            'Please ensure the NCuT has located the test AMWA IS-04 Registry before clicking the \'Next\' button.')
+
+        question = ''.join(paragraphs)
+
+        try:
+            self._invoke_testing_facade(question, [], test_type="action", timeout=600)
+
+        except TestingFacadeException:
+            # pre_test_introducton timed out
+            pass
+
+    def post_tests_message(self):
+        """
+        Controller Test Suite complete!
+        """
+        # Once the tests are complete this post tests message is displayed.
+
+        question = 'NMOS Controller Test Suite complete!\r\n\r\nPlease press the \'Next\' button to exit the tests'
```

Comment:
Are you going to try turning these long `question` strings into multi-line strings?

E.g. something like this, but I suspect it needs tweaking to pass lint and end up with exactly the line endings you want... Some ideas in https://blog.finxter.com/proper-indentation-for-python-multiline-strings/...

```suggestion
        question = """
                   NMOS Controller Test Suite complete!
                   Please press the 'Next' button to exit the tests
                   """
```

Reply:
Interestingly the code as you have it puts a CR at the beginning of the multi-line string. Solved by putting the first line of text on the same line as the triple quotes:
`  question = """NMOS Controller Test Suite complete!`
which works, but looks less nice in the code.

I think it does help to make the pre_tests_message consistent (and remove the embedded \n )

Reply:
How about this:

```python
question = """\
           Meow
           Purr
           """
```

Or:

```python
question = """
           Hiss
           Yowl
           """.strip()
```

Or:

```python
question = textwrap.dedent("""
           Chirp
           Burble
           """)
```

Reply:
Multi-line strings now used where appropriate